bin/radical-pilot-service-wrapper

```diff
@@ -0,0 +1,309 @@
+                continue
+
+        # find pattern
+        start = time.time()
+        regex = re.compile('^.*(%s).*$' % pattern)
+        data  = ''
+
+        log.debug('watching: %s for pattern [%s]' % (fname, pattern))
+        if fstream:
+            log.debug('watching: %s' % fstream)
+            while proc.poll() is None:
+
+                r, _, _ = select.select([fstream], [], [], 0.1)
+                if fstream in r:
+
+                    data += str(os.read(fstream.fileno(), 1).decode())
+
+                    if not data.endswith('\n'):
+                        continue
+
+                    line = data[:-1]
+                    data = ''
+
+                    if verbose:
+                        log.debug('line: %s' % line)
+
+                    match = regex.match(line)
+                    if match:
+                        match = match.groups()[-1]
+                        break
+
+                if timeout and time.time() - start > timeout:
+                    raise ValueError('timeout')
+
+    except Exception as e:
+        err_msg = repr(e)
+        raise
+
+    finally:
+
+        # stop the watcher
+        if timeout_event.is_set():
+            err_msg = 'timeout'
+        else:
+            watch_event.set()
+
+        if verbose:
+            log.debug('stop watcher: %s' % err_msg)
+
+        if proc:
+            ret = proc.poll()
+            if ret is not None and ret > 0:
+                err_msg = 'command failed: %s' % ret
+
+    time.sleep(1)
+
+    # send the result to the agent
+    log.debug('send result: %s [%s]' % (match, err_msg))
+    msg = {'cmd': 'service_info',
+           'arg': {'uid'  : uid,
+                   'error': err_msg,
+                   'info' : match}}
+
+    log.debug(pprint.pformat(msg))
+    pub.put(CHANNEL, msg)
+
+    # register to control pubsub and wait for (hb) messages
+    ctrl_addr_sub = reg.get('bridges.%s' % CHANNEL)['addr_sub']
+
+    last_msg = time.time()
+
+    def _ctrl_cb(topic, msg):
+        global last_msg
+        last_msg = time.time()
+
+    ru.zmq.Subscriber(CHANNEL, ctrl_addr_sub, cb=_ctrl_cb, topic=CHANNEL)
+
+    # now just run until `cmd` completes and return its exit code
+    while proc.poll() is None:
+        log.debug('wait: %s' % proc.poll())
+        if time.time() - last_msg > 10:
+            log.debug('timeout: %s' % proc.poll())
+            log.debug('kill %d' % proc.pid)
+            log.debug(ru.sh_callout('kill %d' % proc.pid))
+            break
+        time.sleep(1.0)
+
+    ret = proc.poll()
+
+    log.debug('done: %s' % ret)
+
+    sys.exit(ret)
+
+
+# ------------------------------------------------------------------------------
+#
+def rank_n():
+
+    global last_msg
+
+    log = ru.Logger('radical.pilot.service_wrapper',
+                    targets='./%s.log' % os.environ['RP_TASK_ID'])
+    log.debug('starting rank %s' % os.environ['RP_RANK'])
+
+    parser = argparse.ArgumentParser(description='')
```

Comment:
```
usage: radical-pilot-service-wrapper [-h] [-c COMMAND] [-o OUTPUT] [-e ERROR]
                                     [-u UID]
radical-pilot-service-wrapper: error: unrecognized arguments: -v
```

Reply:
Duh!  So, I reverted the last change and the check for rank and pattern are now done in `main()` again.  Thanks!